### DISCUSSION TITLE
Is Your waybar Okay?

### DISCUSSION THREAD

**Discussion by kRHYME7** [prasanthrangan/hyprdots#624] [Q&A] [answered]
My waybar  won't start for several minutes so it will reach a certain timeout. After a couple of minutes I can launch it with no problem. Happens After boot. 

error on timeout 

11:44:40.798] [error] Error calling StartServiceByName for org.freedesktop.portal.Desktop: Timeout was reached

**Comment by ghost** [comment]
My waybar is loading perfectly fine. I don't know what could be happening, i believe that's an error saying xdg-desktop-portal hasn't started yet and waybar is unable to start without it. Can you run `systemctl status --user xdg-desktop-portal.service` and see if that gives you anything?

**Reply by kRHYME7** [reply]
Thanks !  I'll try to investigate this. 

```
❯ systemctl status --user xdg-desktop-portal.service
○ xdg-desktop-portal.service - Portal service
     Loaded: loaded (/usr/lib/systemd/user/xdg-desktop-portal.service; static)
     Active: inactive (dead)

Dec 06 13:16:16 ArchLinux systemd[530]: Starting Portal service...
Dec 06 13:16:17 ArchLinux systemd[530]: Started Portal service.

```

**Comment by kRHYME7** [comment, marked as answer]
Thanks Again! 

I don't know what happened but got this error if I ran the xdg-desktop-portal-hyprland
``` ❯ /usr/lib/xdg-desktop-portal-hyprland
/usr/lib/xdg-desktop-portal-hyprland: error while loading shared libraries: libsdbus-c++.so.1: cannot open shared object file: No such file or directory ```


I guess the sdbus-cpp is faulty/ missing. Then I removed the xdg-desktop-portal-hyprland-git package and installed xdg-desktop-portal-hyprland and sdbus-cpp is bundled with it. Now the issue is fixed I guess.

**Reply by ghost** [reply]
Oh i think i know what happened, if you haven't updated xdg-desktop-portal-hyprland-git package in a while, the shared libraries which are in the official repos could be updated leaving xdph behind, if you just recompiled xdg-desktop-portal-hyprland-git, it most likely would have been fixed.

If you use yay, remember that it doesn't update -git packages on it's own, so you have to do `yay -Syu --devel` to check for -git updates. Using paru checks for -git updates on it's own however.

Anyway glad you got it figured out!

**Reply by kRHYME7** [reply]
So Strange because I am using paru -Syu just to install something. Thanks for the tip!

**Reply by ghost** [reply]
Okay i see now, comments in the [xdph aur](https://aur.archlinux.org/packages/xdg-desktop-portal-hyprland-git) are requesting [sdbus](https://archlinux.org/packages/extra/x86_64/sdbus-cpp) be added as a dependency rather than just make-dependency since it's something that's causing issues for people.

So nevermind it wasn't a problem with updates, just the aur package haha.
